src/Adapter/Presenter/Product/ProductLazyArray.php

```diff
@@ -574,18 +595,59 @@ private function shouldShowPrice(
      *
      * @param array $product
      *
-     * @return mixed
+     * @return bool
      */
-    private function shouldShowAddToCartButton($product)
+    private function shouldShowAddToCartButton($product): bool
     {
         return (bool) $product['available_for_order'];
     }
 
-    private function fillImages(
-        ProductPresentationSettings $settings,
-        array $product,
-        Language $language
-    ) {
+    /**
+     * @param array $product
+     *
+     * @return bool
+     */
+    private function shouldShowOutOfStockLabel(array $product): bool
```

Comment:
I think this configuration would be a nice candidate to add in `ProductPresentationSettings` this is where most display configuration is stored and it can also be used in Front which is quite convenient

Reply:
You should add this into the `ProductPresentationSettings` and init its value in `ProductPresenterFactory::getPresentationSettings` This will be much more performant since the value only needs to be set once

With your current implementation each the method is called multiple request are performed again and again